### PR TITLE
feat: deployment since

### PR DESCRIPTION
**Pull request by zhangsoledad**
### What problem does this PR solve?

get_deployments_info rpc adds the returned period, threshold, and epoch which the current state started `since`.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

**Comment by zhangsoledad** [comment]
https://github.com/nervosnetwork/rfcs/pull/403

**Review by eval-exec — COMMENTED** [review]
Please rebased on `develop` branch.

**Review by eval-exec — APPROVED** [review]

File: rpc/README.md [hunk 1]
```diff
@@ -5653,7 +5659,11 @@ An object containing various state info regarding deployments of consensus chang
 
 *   `min_activation_epoch`: [`EpochNumber`](#type-epochnumber) - specifies the epoch at which the softfork is allowed to become active.
 
-*   `state`: [`DeploymentState`](#type-deploymentstate) - With each epoch and softfork, we associate a deployment state. The possible states are
+*   `period`: [`EpochNumber`](#type-epochnumber) - the length in epochs of the signalling period
+
+*   `threshold`: [`DeploymentState`](#type-deploymentstate) - With each epoch and softfork, we associate a deployment state. The possible states are
```
Comment (doitian) [inline rpc/README.md:5664]: Not a complete sentence.

File: util/jsonrpc-types/src/info.rs [hunk 2]
```diff
@@ -55,8 +55,14 @@ pub struct DeploymentInfo {
     pub timeout: EpochNumber,
     /// specifies the epoch at which the softfork is allowed to become active.
     pub min_activation_epoch: EpochNumber,
+    /// the length in epochs of the signalling period
+    pub period: EpochNumber,
+    /// the ratio of blocks with the version bit set required to activate the feature
```
Comment (quake) [inline util/jsonrpc-types/src/info.rs:60]: it looks like this line comment is not generated in the rpc README doc.

File: spec/src/versionbits/mod.rs [hunk 3]
```diff
@@ -273,6 +273,59 @@ pub trait VersionbitsConditionChecker {
...
+        let period = self.period();
+
+        if active_mode == ActiveMode::Always {
+            return Some(0);
+        }
+
+        if active_mode == ActiveMode::Never {
+            return Some(0);
+        }
+
```
Comment (eval-exec) [inline spec/src/versionbits/mod.rs:294]: How about this:
```rust
        match self.active_mode() {
            ActiveMode::Always | ActiveMode::Never => return Some(0),
            _ => {}
        };
```

File: rpc/README.md (hunk 1 above)
Comment (eval-exec) [reply rpc/README.md:5664]: This code changes is based on rust-toolchain `1.61.0`, I guess there may some cargo doc bugs on `1.61.0`, how about try to rebuild rpc docs on `1.67.1` by rebase branch on `develop`.

**Review by eval-exec — APPROVED** [review]

File: spec/src/versionbits/mod.rs (hunk 3 above)
Comment (zhangsoledad) [reply spec/src/versionbits/mod.rs]:
```
      if matches!(self.active_mode(), ActiveMode::Always | ActiveMode::Never) {
            return Some(0);
        }
```

Comment (zhangsoledad) [reply spec/src/versionbits/mod.rs]:
```
      if matches!(self.active_mode(), ActiveMode::Always | ActiveMode::Never) {
            return Some(0);
        }
```

**Pull request merged**